src/main/java/com/example/demo/controllers/ItemController.java

```diff
@@ -108,10 +112,12 @@ public ItemModel getItemFeatured() throws AuctionAppException{
 	
 	@ApiOperation(value = "Enables searching items by name and filtering by different parameters", notes = "Public access")
 	@GetMapping("/api/items/search")
-	public Collection<ItemModel> findItem(@RequestParam String term,@RequestParam List<Integer> categories,
-										  @RequestParam int page,@RequestParam int count,@RequestParam(required = false, defaultValue = "default") ItemSorting sort)
-										  throws AuctionAppException{
-		return imageService.loadImagesForItems(itemService.findItemsValidFilterCategories(term,categories,new SortingPaginationParams(page,count,sort)));
+	public Collection<ItemModel> findItem(@RequestParam String term,@RequestParam List<Integer> categories, @RequestParam List<Integer> subcategories,
+										  @RequestParam(required = false) BigDecimal minPrice,
+										  @RequestParam(required = false) BigDecimal maxPrice,
+										  @RequestParam int page,@RequestParam int count,
+										  @RequestParam(required = false, defaultValue = "default") ItemSorting sort)throws AuctionAppException{
+		return imageService.loadImagesForItems(itemService.findItemsValidFilterCategoriesSubcaetgoriesPrice(term,categories,subcategories,minPrice,maxPrice,new SortingPaginationParams(page,count,sort)));
```

Comment:
let's put these into a single class, this is messy

FilterItemsRequest {
 private term;...